src/coreclr/src/jit/gentree.h

```diff
@@ -4506,6 +4508,8 @@ struct GenTreeCall final : public GenTree
     unsigned char gtCallType : 3;   // value from the gtCallTypes enumeration
     unsigned char gtReturnType : 5; // exact return type
 
+    CorInfoUnmanagedCallConv unmgdCallConv;
```

Comment:
That is an important change, it increases the size of `GenTreeCall` node and it increases the size of all large nodes. Jit typically spends 30% of used memory for trees, do not remember exactly what part of them are large, but this change could be a significant regression for memory consumption. Can it be avoided?

Reply:
Wow, I completely missed this - thanks for catching it @sandreenko! I think we really need to rethink this.

Reply:
There is a pointer to information about tail calls that is lazily allocated when needed: https://github.com/dotnet/runtime/blob/598426c8ccf2d1d41038d711f2187899d50ff367/src/coreclr/src/jit/gentree.h#L3954
It already has what you need I believe. Presumably you could just rename this to `CallSiteInfo` and allocate it when you need it in this scenario as well.

Reply:
By the way, in #341 I also had to add info to `GenTreeCall` and refactored to be able to do this without increasing the size of `GenTreeCall`. At the time I was dreaming of having some hash map on the side in the JIT that allowed attaching arbitrary annotations to nodes for cases like this, where only very few nodes will need the annotation. That would also allow to get rid of the 8 bytes for `tailCallInfo` in the vast majority of calls.

Reply:
That looks like it should work. Thanks for the pointer!

Reply:
> Thanks for the pointer

😄

Reply:
I still need to calculate the unmanaged calling convention in advance to make using the new extensible calling convention system performant, so I can't just use `tailCallInfo` directly. However, since calls to native cannot be tail-called, I can use a union here since the two entries will be used in mutually exclusive scenarios.